scene/resources/default_theme/default_theme.cpp

```diff
@@ -256,7 +256,7 @@ void fill_default_theme(Ref<Theme> &theme, const Ref<Font> &default_font, const
 	theme->set_color("font_outline_color", "OptionButton", Color(1, 1, 1));
 
 	theme->set_constant("h_separation", "OptionButton", 2 * scale);
-	theme->set_constant("arrow_margin", "OptionButton", 4 * scale);
+	theme->set_constant("arrow_margin", "OptionButton", 0);
```

Comment:
This technically breaks compatibility. I think it's fine to leave it; I tested the old value and the gap is small after bugfixes.